### ISSUE TITLE
[CS1.6] Steam Networking Servers no longer listed in the master server list

### ISSUE THREAD

**Issue by Splatt581** [ValveSoftware/halflife#3782]
Some time ago, the Steam Networking listen servers (with `169.254` fakeip subnet) stopped being displayed in server browser - the master server stopped sending them in the general list of servers.

![Untitled](https://github.com/ValveSoftware/halflife/assets/31417200/f865bf62-6844-4cb6-bcbe-b090618a7a99)

This problem is not observed in other GoldSrc games. I believe the problem arose due to the large number of servers in CS 1.6, now more than 20k of them have been added to the master server due to fake redirect servers.

**Comment by 0Ky** [comment]
@Splatt581 It works fine for me, so I'm unable to reproduce the issue you've mentioned. Does it occur occasionally? It might be that no players were hosting using the new "Steam Networking" feature in CS 1.6 at the time. I suggest you also verify using the following web API for discrepancies: `https://api.steampowered.com/IGameServersService/GetServerList/v1/?key=YOUR_API_KEY&filter=appid\10`, servers with FakeIPs should be listed from the beginning.

@that-is-not-great No need to hijack this issue thread for your unrelated problem, which is already being tracked at #2064.

**Comment by Splatt581** [comment]
> @Splatt581 It works fine for me, so I'm unable to reproduce the issue you've mentioned. Does it occur occasionally? It might be that no players were hosting using the new "Steam Networking" feature in CS 1.6 at the time. I suggest you also verify using the following web API for discrepancies: `https://api.steampowered.com/IGameServersService/GetServerList/v1/?key=YOUR_API_KEY&filter=appid\10`, servers with FakeIPs should be listed at the beginning.

Yeah, I had previously used this webapi endpoint for a test and the steam networking servers were there. I am inclined to believe that my serverbrowser stopped receiving steam networking servers when many farmers fake servers appeared on the master server (more than 20k in total).

**Comment by laimiukas3** [comment]
what is this nonsense in steam masterserver
https://www.youtube.com/watch?v=BKsYHKEPivs&t=69s

**Comment by Maxi605** [comment]
> > what is this nonsense in steam masterserver https://www.youtube.com/watch?v=BKsYHKEPivs&t
> 
> yea ridiculous. valve is too lazy to fix anything there. btw add column IP and you see they are all fake redirect servers with the same IP. to valve: add a max. ports per IP limit and/or add a in-game client filter, max servers per IP: 5, 10, 50, etc similar to ping filter

This wouldn't really fix the problem and it isn't ideal, not all servers sharing the same IP are fake servers. The issue should be tackled differently.

**Comment by Splatt581** [comment]
Since the beginning of April, Valve changed something in the issuance of the list of game servers by the Master server. Now I get Steam networking servers, so I no longer have this problem. Also, now the Steam web api `IGameServersService/GetServerList` does not give Steam networking servers at the beginning of the list.

![Image](https://github.com/user-attachments/assets/81289e4c-7dd1-4977-a474-7ab72ea6256d)
